libtiledbvcf/src/utils/utils.cc

```diff
@@ -314,17 +314,27 @@ uint64_t ceil(uint64_t x, uint64_t y) {
 std::mutex cfg_mutex;
 std::mutex ctx_mutex;
 std::mutex init_mutex;
+
+// Store config and context in unique_ptr so we don't leak
+std::unique_ptr<tiledb::Config> hfile_tiledb_vfs_config_ptr = nullptr;
+std::unique_ptr<tiledb::Context> hfile_tiledb_vfs_ctx_ptr = nullptr;
 void set_htslib_tiledb_config(const std::vector<std::string>& tiledb_config) {
   const std::lock_guard<std::mutex> lock(cfg_mutex);
-  tiledb_error_t* err;
-  tiledb_config_alloc(&hfile_tiledb_vfs_config, &err);
-  tiledb::impl::check_config_error(err);
-  set_tiledb_config(tiledb_config, hfile_tiledb_vfs_config);
+  if (hfile_tiledb_vfs_config_ptr != nullptr) {
```

Comment:
Thanks for catching this!

Reply:
Follow up question then: this setup seems to imply that the user can only set the config on the htslib once, corresponding to the config passed the first time a `Dataset` is constructed? That's probably OK (usually only use one config), but it could be surprising.

Don't need to do it now, but might want to give a warning or error if someone tries to set a different config here, to avoid surprise.

Reply:
I put a story into core to provide an comparator for now I just implemented the comparison we have in MariaDB. We now will update the config/context only if the config is different. If the config isn't different, no change is made.

Now this does introduce a possible thread-safety issue of switching the config/context on htslib but in TileDB-VCF this isn't so much a problem. Concurrent queries with different configs are going to be extremely unlikely. More likely is a user just in a single python sessions changing parameters slowly over time which is fine.

Reply:
👍